packages/docusaurus-theme-classic/src/theme/hooks/useTabGroupChoice.js

```diff
@@ -0,0 +1,44 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import {useState, useCallback, useEffect} from 'react';
+
+const useTabGroupChoice = () => {
+  const [choices, setChoices] = useState({});
+  const setChoiceSyncWithLocalStorage = useCallback(
+    newChoices => {
+      try {
+        localStorage.setItem('tab-group', JSON.stringify(newChoices));
```

Comment:
First, put storage value in separate const outside the hook-level. Second, add a namespace to that, for example `docusaurus.tab-group`.

Reply:
> put storage value in separate const outside the hook-level.

:up: @SamChou19815

Reply:
@lex111 Do you mean moving the literal`'docusaurus.tab-choice'` outside of the hooks?

Reply:
Something like this (come up with a better name for const):

```js
const STORAGE_KEY = ...

const setChoiceSyncWithLocalStorage = ...
```